### PR TITLE
V8 transport upgrade guide

### PR DESCRIPTION
**Pull request by timbussmann**
TODO:
* [x] Add links to the specific transport upgrade guides

### CODE REVIEW

File: nservicebus/upgrades/7to8/index.md [hunk 1]
```diff
@@ -11,6 +11,38 @@ upgradeGuideCoreVersions:
...
+
+var routing = endpointConfiguration.UseTransport(transport);
+routing.RouteToEndpoint(typeof(MyMessage), "DestinationEndpoint");
+```
+
+The transport can be configured via the transport's configuration type which derives from `TransportDefinition`. See the [transport upgrade guide](/nservicebus/upgrades/7to8/transport.md) for further details.
+
+## Connection strings
+
+Configuring a transport's connection using `.ConnectionStringName(name)`, which was removed for .NET Core in NServiceBus version 7, has been removed all platforms in NServiceBus version 8. To continue to retrieve the connection string by the named value in the configuration, first retrieve the connection string and then pass it to the `.ConnectionString(value)` configuration.
```
Comment (mauroservienti) [inline nservicebus/upgrades/7to8/index.md:41]:
```suggestion
Configuring a transport's connection using `.ConnectionStringName(name)`, which was removed for .NET Core in NServiceBus version 7, has been removed from all platforms in NServiceBus version 8. To continue to retrieve the connection string by the named value in the configuration, first retrieve the connection string and then pass it to the `.ConnectionString(value)` configuration.
```

Isn't `ConnectionString` gone?

File: nservicebus/upgrades/7to8/index.md [hunk 2]
```diff
@@ -134,3 +159,15 @@ The following transports might need migration:
...
 * SQS
+
+## Outbox configuration
+
+NServiceBus version 8 requires explicit configuration of the transport transaction mode when using [Outbox](/nservicebus/outbox/):
+
+```csharp
+var transport = endpointConfiguration.UseTransport<MyTransport>();
+
+transport.TransportTransactionMode = TransportTransactionMode.ReceiveOnly;
```
Comment (mauroservienti) [inline nservicebus/upgrades/7to8/index.md:170]: Does it make sense to specify, if I understood correctly, that `ReceiveOnly` is the only supported mode when using the Outbox?

File: nservicebus/upgrades/7to8/transport.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ - 7
+ - 8
+---
+
+## Transport Specific Sonfiguration
+
+See the transport specific upgrade guides for further details on the configuration options:
+
+* [Azure Service Bus transport upgrade guide](/transports/upgrades/asbs-1to2.md)
+* [Azure Storage Queues transport upgrade guide](TODO)
```
Comment (mauroservienti) [inline nservicebus/upgrades/7to8/transport.md:17]:
```suggestion
* [Azure Storage Queues transport upgrade guide](/transports/upgrades/asq-9to10.md)
```

**Review by mauroservienti — CHANGES_REQUESTED** [review]

File: nservicebus/upgrades/7to8/index.md [hunk 4]
```diff
@@ -134,3 +159,15 @@ The following transports might need migration:
 * Azure Storage Queues
 * SQL Transport
 * SQS
+
+## Outbox configuration
+
+NServiceBus version 8 requires explicit configuration of the transport transaction mode when using [Outbox](/nservicebus/outbox/):
```
Comment (mauroservienti) [inline nservicebus/upgrades/7to8/index.md:165]:
```suggestion
NServiceBus version 8 requires the transport transaction mode to be explicitly set to `ReceiveOnly` when using [Outbox](/nservicebus/outbox/):
```

@tmasternak thoughts 👆🏻 ?

File: nservicebus/upgrades/7to8/index.md [hunk 5]
```diff
@@ -11,6 +11,38 @@ upgradeGuideCoreVersions:
...
+    TransportTransactionMode = TransportTransactionMode.ReceiveOnly
+};
+
+var routing = endpointConfiguration.UseTransport(transport);
+routing.RouteToEndpoint(typeof(MyMessage), "DestinationEndpoint");
+```
+
+The transport can be configured via the transport's configuration type which derives from `TransportDefinition`. See the [transport upgrade guide](/nservicebus/upgrades/7to8/transport.md) for further details.
+
+## Connection strings
```
Comment (timbussmann) [inline nservicebus/upgrades/7to8/index.md:39]: should this section be moved to the transport specific page or do we keep this here?

**Comment by timbussmann** [comment]
added some notes about the backwards-compatibility

File: nservicebus/upgrades/7to8/index.md (hunk 5 above)
Comment (mauroservienti) [reply nservicebus/upgrades/7to8/index.md:41]: I think so, especially because there are transports for which a connection string doesn't make sense.

**Review by SzymonPobiega — APPROVED** [review]

File: nservicebus/upgrades/7to8/transport.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+## Transport Specific Configuration
+
+See the transport specific upgrade guides for further details on the configuration options:
+
+* [Azure Service Bus transport upgrade guide](/transports/upgrades/asbs-1to2.md)
+* [Azure Storage Queues transport upgrade guide](/transports/upgrades/asq-9to10.md)
+* [RabbitMQ transport upgrade guide](/transports/upgrades/rabbitmq-6to7.md)
+* [MSMQ transport upgrade guide](/transports/upgrades/msmq-1to2.md)
+* [SQL Server transport upgrade guide](TODO)
```
Comment (timbussmann) [inline nservicebus/upgrades/7to8/transport.md:39]: can we merge https://github.com/Particular/docs.particular.net/pull/5193 so we can go ahead with this?

File: nservicebus/upgrades/7to8/transport.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+var routing = endpointConfiguration.UseTransport(transport);
+routing.RouteToEndpoint(typeof(MyMessage), "DestinationEndpoint");
+```
+
+## Transport Specific Configuration
+
+See the transport specific upgrade guides for further details on the configuration options:
+
+* [Azure Service Bus transport upgrade guide](/transports/upgrades/asbs-1to2.md)
+* [Azure Storage Queues transport upgrade guide](/transports/upgrades/asq-9to10.md)
```
Comment (danielmarbach) [inline nservicebus/upgrades/7to8/transport.md:36]:
```suggestion
* [Azure Storage Queues transport upgrade guide](/transports/upgrades/asq-10to11.md)
```

Comment (danielmarbach) [reply nservicebus/upgrades/7to8/transport.md:36]: @timbussmann we have released a new major that targets v7

**Review by tmasternak — APPROVED** [review]

**Review by mauroservienti — APPROVED** [review]

**Pull request merged**